src/router.ts

```diff
@@ -33,11 +33,18 @@ export type OneSchemaRouterConfig<R extends Router<any, any>> = {
 
 export type NamedClient<Schema extends ZodSchema> = {
   [Route in keyof Schema as Schema[Route]['name']]: (
-    request: z.infer<Schema[Route]['request']> & PathParamsOf<Route>,
+    request: z.input<Schema[Route]['request']> & PathParamsOf<Route>,
```

Comment:
This change (and the one below) ensures that we use the correct inference method for typing the request objects between the `client(...)` helper and the implementation function.

The client helper should get the "input" type (i.e. the type that the schema accepts), and the implementation function should get the "output" type (i.e. the type that the schema produces).

You can read more about input vs output here: https://zod.dev/?id=zodtype-with-zodeffects